admin/server/magic_tokens.go

```diff
@@ -242,6 +242,19 @@ func magicAuthTokenToPB(tkn *database.MagicAuthTokenWithUser) (*adminv1.MagicAut
 		}
 	}
 
+	proj, err := s.admin.DB.FindProject(ctx, tkn.ProjectID)
+	if err != nil {
+		if errors.Is(err, database.ErrNotFound) {
+			return nil, status.Error(codes.NotFound, fmt.Sprintf("project with id %s not found", tkn.ProjectID))
+		}
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	org, err := s.admin.DB.FindOrganization(ctx, proj.OrganizationID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to find organization for project: %v", err.Error())
+	}
+
```

Comment:
instead of doing db queries here, we can change the find magic auth token db queries to lookup org name, org custom domain and project name by doing join 3 way join with org and project tables and also we will need to overload the `MagicAuthToken` struct with org name, customDomain and project name, so was not sure. Should we do that instead ?

Reply:
Since it's a relatively infrequent code path, I would just suggest doing the project/org lookup separately in the server handler, and passing them to `magicAuthTokenToPB`. For example, in `ListMagicAuthTokens` it's already doing a project lookup anyway.